### PR TITLE
Transform `AuthLayoutContainer`  component hierarchy to tsx

### PR DESCRIPTION
**Pull request by ghost**
#### Pre-flight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

- chore: update publish debug into to TS
- refactor: component ui from react props to TS types
- chore: update auth layout to TS
- refactor: auth layout container from old HOC to typescript inject annotation
- refactor: routes from prop types to typescipt type
- chore: update eslint config to disable default prop type rule

#### Motivation and Context

- [ Convert to typescript #42 ](https://github.com/ferdium/ferdium-app/issues/42)

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/components/ui/AppLoader/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { Component } from 'react';
+import { Component, ReactElement } from 'react';
 import classnames from 'classnames';
-
-import injectStyle from 'react-jss';
+import withStyles, { WithStylesProps } from 'react-jss';
+import { Theme } from 'src/themes';
```
Comment (vraravam) [inline src/components/ui/AppLoader/index.tsx]: Please always use relative file paths - that's the standard we have been using till date.

File: src/components/auth/AuthLayout.tsx [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
+  MouseEventHandler,
+  ReactElement,
+} from 'react';
 import { observer } from 'mobx-react';
 import { TitleBar } from 'electron-react-titlebar/renderer';
-
-import { injectIntl } from 'react-intl';
+import { injectIntl, WrappedComponentProps } from 'react-intl';
 import { mdiFlash } from '@mdi/js';
+import { GlobalError } from 'src/@types/ferdium-components.types';
```
Comment (vraravam) [inline src/components/auth/AuthLayout.tsx]: Please use relative paths - that's the standard we have been using till date

File: src/containers/auth/AuthLayoutContainer.tsx [hunk 3]
```diff
@@ -39,7 +40,7 @@ class AuthLayoutContainer extends Component<AuthLayoutContainerProps> {
       <ThemeProvider theme={stores.ui.theme}>
         <AuthLayout
           error={globalError.response}
-          pathname={stores.router.location.pathname}
+          // pathname={stores.router.location.pathname} // TODO - [TS DEBT] path name props is not used in AuthLayout
```
Comment (vraravam) [inline src/containers/auth/AuthLayoutContainer.tsx]: If you have verified that this is not being used, please delete the property.

File: src/features/publishDebugInfo/Component.tsx [hunk 4]
```diff
@@ -1,18 +1,15 @@
 import { inject, observer } from 'mobx-react';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { defineMessages, injectIntl } from 'react-intl';
-import injectSheet from 'react-jss';
+import { Component, ReactElement } from 'react';
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import withStyles, { WithStylesProps } from 'react-jss';
+import { StoresProps } from 'src/@types/ferdium-components.types';
```
Comment (vraravam) [inline src/features/publishDebugInfo/Component.tsx]: Please use relative paths.

File: src/features/publishDebugInfo/Component.tsx [hunk 5]
```diff
@@ -159,18 +163,15 @@ class PublishDebugLogModal extends Component {
             <p className={classes.info}>
               {intl.formatMessage(messages.published)}
             </p>
-            <Input showLabel={false} value={`${DEBUG_API}/${log}`} readonly />
+            <Input showLabel={false} value={`${DEBUG_API}/${log}`} />
```
Comment (vraravam) [inline src/features/publishDebugInfo/Component.tsx]: Why remove the `readonly`? That would make this a text box that the user can type into. If this ts-conversion complains, then can we convert this into a label?

**Review by vraravam — CHANGES_REQUESTED** [review]

**Comment by vraravam** [comment]
@muhamedsalih-tw - I have rebased from the latest `develop` branch and force pushed into your branch. As a new commit, I have fixed a couple of the review comments related to relative paths. Since this PR already has a large number of files, please fix the few oustanding review comments and then mark this PR as ready for review. Lets merge in small chunks please

File: src/containers/auth/AuthLayoutContainer.tsx (hunk 3 above)
Comment (ghost) [reply src/containers/auth/AuthLayoutContainer.tsx]: I have removed it , will push it

File: src/features/publishDebugInfo/Component.tsx (hunk 5 above)
Comment (ghost) [reply src/features/publishDebugInfo/Component.tsx]: readOnly is allowed by TS as we extending a InputHTMLAttributes props
but it not used inside Input tag 
I shall leave a comment for now, will check it later

File: src/features/publishDebugInfo/Component.tsx (hunk 4 above)
Comment (ghost) [reply src/features/publishDebugInfo/Component.tsx]: will follow relative path format

File: src/components/auth/AuthLayout.tsx (hunk 2 above)
Comment (ghost) [reply src/components/auth/AuthLayout.tsx]: will follow relative path format

File: src/components/ui/AppLoader/index.tsx (hunk 1 above)
Comment (ghost) [reply src/components/ui/AppLoader/index.tsx]: will follow relative path format

**Review by vraravam — APPROVED** [review]
LGTM - tested locally with pre-existing profile (no server)

**Comment by vraravam** [comment]
@all-contributors please add @muhamedsalih-tw for code

**Pull request merged**